### PR TITLE
Pcre2 3194 v2

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3194

Describe changes:
- Use PCRE2 instead of pcre for transform pcrexform
- qa : remove some obsolete files

This draft PR is about testing PCRE2 support by the CI
It obviously needs improvements

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -637,6 +637,20 @@
         exit 1
     fi
 
+    PCRE2=""
+    #AC_CHECK_HEADER(pcre2.h,,[AC_MSG_ERROR(pcre2.h not found ...)])
```
Comment (catenacyber) [inline configure.ac:641]: On my machine, this fails even if I have pcre2.h in my /usr/local/include directory

```
checking pcre.h usability... yes
checking pcre.h presence... yes
checking for pcre.h... yes
checking for pcre_get_substring in -lpcre... yes
checking pcre2.h usability... no
checking pcre2.h presence... no
checking for pcre2.h... no
configure: error: pcre2.h not found ...
```

Any idea why ? (it is a symbolic link unlike pcre.h)

Comment (victorjulien) [reply configure.ac:641]: You `config.log` should hold some more info.

Comment (catenacyber) [reply configure.ac:641]: Thanks, my PCRE2 version asks `PCRE2_CODE_UNIT_WIDTH` to be defined before including pcre2.h

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/5981

**Pull request closed without merging**